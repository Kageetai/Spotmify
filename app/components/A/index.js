/**
 * A link to a certain page, an anchor tag
 */

import styled from 'styled-components';

import { colors } from 'global-styles';

const A = styled.a`
  color: ${colors.primary};
  text-decoration: none;

  &:hover {
    color: ${colors.primaryHover};
  }
`;

export default A;
