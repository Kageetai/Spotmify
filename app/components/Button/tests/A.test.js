import React from 'react';
import { mount } from 'enzyme';
import renderer from 'react-test-renderer';

import A from '../A';

describe('<A />', () => {
  it('should match the snapshot', () => {
    const wrapper = renderer.create(<A />).toJSON();
    expect(wrapper).toMatchSnapshot();
  });

  it('should have a className attribute', () => {
    const wrapper = mount(<A />);
    expect(wrapper.find('a').prop('className')).toBeDefined();
  });

  it('should adopt a valid attribute', () => {
    const id = 'test';
    const wrapper = mount(<A id={id} />);
    expect(wrapper.prop('id')).toEqual(id);
  });

  it('should not adopt an invalid attribute', () => {
    const wrapper = mount(<A attribute="test" />);
    expect(wrapper.find('a').prop('attribute')).toBeUndefined();
  });
});
