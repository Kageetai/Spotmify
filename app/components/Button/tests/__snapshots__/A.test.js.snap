// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<A /> should match the snapshot 1`] = `
.c0 {
  display: inline-block;
  box-sizing: border-box;
  padding: 0.25em 2em;
  -webkit-text-decoration: none;
  text-decoration: none;
  border-radius: 4px;
  -webkit-font-smoothing: antialiased;
  -webkit-touch-callout: none;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  cursor: pointer;
  outline: 0;
  font-family: 'Helvetica Neue',Helvetica,Arial,sans-serif;
  font-weight: bold;
  font-size: 16px;
  border: 2px solid #18aa4c;
  color: #18aa4c;
}

.c0:hover {
  color: #249446;
  border-color: #249446;
}

.c0:active {
  background: #18aa4c;
  color: #fff;
}

<a
  className="c0"
/>
`;
